### PR TITLE
homework3 RPVanarx

### PR DESCRIPTION
**Pull request by RPVanarx**

### CODE REVIEW

File: homeworks/roman.plynko_RPVanarx/homework_3/task2.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    input: process.stdin,
+    output: process.stdout,
+});
+
+let searchPowNumber;
+let minValue;
+let maxValue;
+
+function searchPow(n, min, max) {
+    if (n >= min && n <= max) {
```
Comment (markoffden) [inline homeworks/roman.plynko_RPVanarx/homework_3/task2.js]: 1. make you function to take only one argument (number to check), and handle the rest with a cycle, now you pass min and max values  each time, but these values never change

2. also keep all data you need for function execution inside function body, unless they have specific purpose to be in outer scope

this code may work, but it needs decent optimization

File: homeworks/roman.plynko_RPVanarx/homework_3/task2.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        maxValue = +maxValue;
+        if ( // input check
+            Number.isInteger(searchPowNumber)
+            && Number.isInteger(minValue)
+            && Number.isInteger(maxValue)
+            && searchPowNumber > 1
+            && minValue < maxValue
+        ) { // if true
+            rl.close();
+            searchPow(searchPowNumber, minValue, maxValue);
```
Comment (markoffden) [inline homeworks/roman.plynko_RPVanarx/homework_3/task2.js]: check previous comment, using loop will be much more readable here

File: homeworks/roman.plynko_RPVanarx/homework_3/task1.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    const newmass = [];
+    for (let i = 0; i <= max; i += step) {
+        newmass.push(i);
+    }
+    return newmass;
+}
+
+const summ = massCreate(4, 32).map(pow).reduce((a, b) => a + b); // task 1.3 & 1.4
+
+console.log(summ);
```
Comment (markoffden) [inline homeworks/roman.plynko_RPVanarx/homework_3/task1.js]: all good here

File: homeworks/roman.plynko_RPVanarx/homework_3/task2.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    input: process.stdin,
+    output: process.stdout,
+});
+
+(function create() {
+    let searchPowNumber;
+    let minValue;
+    let maxValue;
+
+    function searchPow(number) {
```
Comment (markoffden) [inline homeworks/roman.plynko_RPVanarx/homework_3/task2.js]: this function should not interact with outer variables, you script should contain only 2 elements:

1. recursive function taking one argument
2. the loop from min to max number loggin all number matching the condition

File: homeworks/roman.plynko_RPVanarx/homework_3/task2.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        [searchPowNumber, minValue, maxValue] = value.split(' ').map(Number);
+        if (minValue > 1 && searchPowNumber > 1 && minValue < maxValue) {
+            rl.close();
+            searchPow(searchPowNumber);
+        } else { // repeat input
+            console.log('Opps you have entered an invalid value, try again');
+            create(); // recursion
+        }
+    });
+}()); // function call "create"
```
Comment (markoffden) [inline homeworks/roman.plynko_RPVanarx/homework_3/task2.js]: wrapping everything in self-invoking function not needed, your recursive function shoul be call inside the loop to check the number

**Comment by RPVanarx** [comment]
I just wanted to do more interesting task.  Done, check please .

**Comment by markoffden** [comment]
fixed, good job

**Pull request merged**